### ISSUE TITLE
PrettyDiff: 1 more end type than start types.

### ISSUE THREAD

**Issue by aaronmw** [panoply/vscode-liquid#66]
Figured I'd make this its own issue, though seems related to #42 

The error:

`[1/25/2021, 9:51:32 PM] PrettyDiff: 1 more end type than start types.`

On every save. I swear it was formatting this document fine until... Something. And now this. I don't see anything wrong with my document 🤔

**Comment by jakobsturm** [comment]
Hi, I am experiencing the same issue! Even if I just use plain HTML without any liquid syntax it crashes.

**Comment by panoply** [comment]
This will likely cause due to self closing HTML tags, for example:

```html
<img src="/path" />
```
You need explicitly set a ruleset for this. Use a configuration like below:

```json
{
  "ignore": [
    {
      "type": "liquid",
      "begin": "comment",
      "end": "endcomment"
    }
  ],
  "html": {
    "correct": true,
    "force_attribute": false,
    "braces": false,
    "preserve": 1,
    "wrap": 80,
    "preserve_text": true,
    "content": true,
    "space_close": true,
    "tags": [
      {
        "tag": "icon",
        "rules": {
          "force_attribute": false,
          "space_close": true
        }
      },
      {
        "tag": "source",
        "rules": {
          "force_attribute": true,
          "space_close": true
        }
      }
    ]
  },
  "js": {
    "preserve": 1,
    "method_chain": 0,
    "quote_convert": "none",
    "format_array": "indent",
    "format_object": "indent",
    "braces": false,
    "no_semicolon": false,
    "brace_block": true
  },
  "scss": {
    "css_insert_lines": true,
    "preserve": 2,
    "braces": false,
    "brace_block": true
  },
  "css": {
    "css_insert_lines": true,
    "preserve": 2,
    "braces": false,
    "brace_block": true
  },
  "json": {
    "preserve": 0,
    "braces": true,
    "brace_block": false,
    "no_semicolon": true
  }
}

``` 

Again, these are addressed in Liquify #56

**Comment by panoply** [comment]
🚢  Shipped [v3.0.0](https://github.com/panoply/vscode-liquid/releases/tag/v3.0.0)
